lib/googleauth/user_authorizer.rb

```diff
@@ -228,6 +248,23 @@ def store_credentials user_id, credentials
         credentials
       end
 
+      # The code verifier for PKCE for OAuth 2.0. When set, the
+      # authorization URI will contain the Code Challenge and Code
+      # Challenge Method querystring parameters, and the token URI will
+      # contain the Code Verifier parameter.
+      #
+      # @param [String|nil] new_code_erifier
+      def code_verifier= new_code_verifier
+        @code_verifier = new_code_verifier
+      end
+
+      # Generate the code verifier needed to be sent while fetching
+      # authorization URL.
+      def generate_code_verifier
```

Comment:
Should this be a class method instead of an instance method? If it's an instance method, it kinda implies it might set `@code_verifier` which it doesn't.

Reply:
I was also thinking which will be a better implementation. I'll go ahead and set the `@code_verifier` in the `generate_code_verifier` method, and keep the method as an instance method only.